### DISCUSSION TITLE
Table of content on an article

### DISCUSSION THREAD

**Discussion by TechWatching** [statiqdev/Discussions#97] [Q&A] [answered]
I am using Statiq with Clean blog theme and I saw in the documentation that it could be possible for each post to have a table of content of the post based on its headlines. The idea would be to display this toc at the beginning of the article or on the side of the article. Does somebody know where I could find some example of doing such a thing ?

**Comment by TheBrambleShark** [comment, marked as answer]
When I was playing with this, I ended up doing this through some [razor code](https://github.com/LuzFaltex/luzfaltex.com/blob/main/input/Shared/_tocList.cshtml).

It never ended up in my final project, but the debugging class that shows how I did it still exists. Also, be sure to set `GatherHeadingsLevel` to a value of higher equal to the deepest header (h1 being 1, h2 being 2, and so on) in your global config.

I should also point out that this only works with markdown pages. Razor pages are not supported since not all of them are valid html documents on their own. Some are C# scripts or partials.

See also #58

**Reply by daveaglick** [reply]
Great answer, thanks @Foxtrek64! Yep, there's two things to getting this to work:
- The first is to set `GatherHeadingsLevels` to whatever depth you want (by default it will gather only the top-level headings if you don't change the setting).
- The second is to get the headings from the document in your Razor code: `Document.GetDocumentList(HtmlKeys.Headings)`. The return of that is a list of nested documents that themselves contain information about the heading. @Foxtrek64 example is a great starting point, but feel free to ask for more help if it's not totally clear from that.
